### PR TITLE
refactor(region): Splitting of KSA(Saudi Arabia) Region from ERPNext

### PR DESCRIPTION
**Pull request by sankarsubramaniankvs**
This Pull Request Will remove the code related to Saudi Arabia Region from ERPNext. It will come in a separate app that will be published in the frappe marketplace and will be used only by companies that need KSA Compliance. The idea behind it is to remove code unnecessary for a user. ERPNext should only have a generic code that can be used by everyone.

Link to new app: https://github.com/8848digital/KSA

### CODE REVIEW

File: erpnext/hooks.py [hunk 1]
```diff
@@ -311,15 +311,12 @@
 		"on_submit": [
 			"erpnext.regional.create_transaction_log",
 			"erpnext.regional.italy.utils.sales_invoice_on_submit",
-			"erpnext.regional.saudi_arabia.utils.create_qr_code",
```
Comment (deepeshgarg007) [inline erpnext/hooks.py:314]: Din't find this in the new app

File: erpnext/hooks.py [hunk 2]
```diff
@@ -311,15 +311,12 @@
...
 		],
 		"on_cancel": [
-			"erpnext.regional.italy.utils.sales_invoice_on_cancel",
```
Comment (deepeshgarg007) [inline erpnext/hooks.py:317]: Din't find this in the new app

File: erpnext/hooks.py [hunk 3]
```diff
@@ -311,15 +311,12 @@
...
-			"erpnext.regional.saudi_arabia.utils.delete_qr_code_file",
```
Comment (deepeshgarg007) [inline erpnext/hooks.py:318]: Din't find this in the new app

File: erpnext/hooks.py [hunk 4]
```diff
@@ -311,15 +311,12 @@
...
+			"erpnext.regional.italy.utils.sales_invoice_on_cancel"
 		],
 		"on_trash": "erpnext.regional.check_deletion_permission",
 	},
-	"POS Invoice": {"on_submit": ["erpnext.regional.saudi_arabia.utils.create_qr_code"]},
```
Comment (deepeshgarg007) [inline erpnext/hooks.py:322]: Din't find this in the new app

**Review by deepeshgarg007 — CHANGES_REQUESTED** [review]
Add a depreciation warning on version-14 like https://github.com/frappe/erpnext/pull/30861/files

Add a patch to delete doctypes from old modules https://github.com/frappe/erpnext/pull/30830/files#diff-bdbb775d3a1469cacdfc456d9abbc26cc35c72fdeeeb96d6db481d3856a11abaR1

**Comment by sankarsubramaniankvs** [comment]
> erpnext

I have added the patches

File: erpnext/hooks.py (hunk 1 above)
Comment (sankarsubramaniankvs) [reply erpnext/hooks.py:314]: Have added all required hooks to the new app

File: erpnext/hooks.py (hunk 2 above)
Comment (sankarsubramaniankvs) [reply erpnext/hooks.py:317]: Have added all required hooks to the new app

File: erpnext/hooks.py (hunk 3 above)
Comment (sankarsubramaniankvs) [reply erpnext/hooks.py:318]: Have added all required hooks to the new app

File: erpnext/hooks.py (hunk 4 above)
Comment (sankarsubramaniankvs) [reply erpnext/hooks.py:322]: Have added all required hooks to the new app

File: erpnext/patches/v15_0/saudi_depreciation_warning.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import click
+
+
+def execute():
+
+	click.secho(
+		"Saudi Arabia Region is moved to a separate app and will be removed from ERPNext in version-15.\n"
+		"When upgrading to ERPNext version-15, please install the app to continue using the Education domain: https://github.com/8848digital/KSA",
```
Comment (deepeshgarg007) [inline erpnext/patches/v15_0/saudi_depreciation_warning.py]:
```suggestion
		"When upgrading to ERPNext version-15, please install the app to continue using the KSA VAT features: https://github.com/8848digital/KSA",
```

File: erpnext/patches/v15_0/saudi_depreciation_warning.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import click
+
+
+def execute():
+
+	click.secho(
+		"Saudi Arabia Region is moved to a separate app and will be removed from ERPNext in version-15.\n"
```
Comment (deepeshgarg007) [inline erpnext/patches/v15_0/saudi_depreciation_warning.py]:
```suggestion
		"Regional features for KSA are moved to a separate app and will be removed from ERPNext in version-15.\n"
```

**Review by deepeshgarg007 — CHANGES_REQUESTED** [review]

**Comment by deepeshgarg007** [comment]
2 checks also failing, please fix them

**Comment by barredterra** [comment]
@sankarsubramaniankvs you forgot to clean up the accounting workspace:

https://github.com/frappe/erpnext/blob/c3d455653e717b539a6fc567fcd8ddbdbf5c717f/erpnext/accounts/workspace/accounting/accounting.json#L519-L530

https://github.com/frappe/erpnext/blob/c3d455653e717b539a6fc567fcd8ddbdbf5c717f/erpnext/accounts/workspace/accounting/accounting.json#L1032-L1042

**Comment by sankarsubramaniankvs** [comment]
> 

Hi @barredterra. Thanks for bringing this to my notice. will check this out

**Pull request merged**